File: apps/python/pipeline.py
"""
shipfast-api · Python App
Pipeline module — chains calculator functions together.
Used for integration testing demos.
"""

from calculator import add, subtract, multiply, divide, is_even


def calculate_discount(price, discount_percent):
    """Apply a percentage discount to a price."""
    discount = divide(multiply(price, discount_percent), 100)
    return subtract(price, discount)


def calculate_total(items):
    """Sum a list of item prices."""
    total = 0
    for price in items:
        total = add(total, price)
    return total


def calculate_average(items):
    """Calculate the average of a list of numbers."""
    total = calculate_total(items)
    return divide(total, len(items))


def is_bulk_order(quantity, threshold=10):
    """Return True if quantity qualifies as a bulk order."""
    return quantity >= threshold

File: apps/node/pipeline.js
/**
 * shipfast-api · Node App
 * Pipeline module — chains utility functions together.
 * Used for integration testing demos.
 */

const { add, subtract, multiply, divide, isEven } = require("./utils");

/**
 * Apply a percentage discount to a price.
 * @param {number} price
 * @param {number} discountPercent
 * @returns {number}
 */
function calculateDiscount(price, discountPercent) {
  const discount = divide(multiply(price, discountPercent), 100);
  return subtract(price, discount);
}

/**
 * Sum a list of item prices.
 * @param {number[]} items
 * @returns {number}
 */
function calculateTotal(items) {
  return items.reduce((total, price) => add(total, price), 0);
}

/**
 * Calculate the average of a list of numbers.
 * @param {number[]} items
 * @returns {number}
 */
function calculateAverage(items) {
  if (items.length === 0) {
    throw new Error("Cannot divide by zero");
  }
  return divide(calculateTotal(items), items.length);
}

/**
 * Return true if quantity qualifies as a bulk order.
 * @param {number} quantity
 * @param {number} threshold
 * @returns {boolean}
 */
function isBulkOrder(quantity, threshold = 10) {
  return quantity >= threshold;
}

module.exports = { calculateDiscount, calculateTotal, calculateAverage, isBulkOrder };